### PR TITLE
taus

### PR DESCRIPTION
**Pull request by daiktas**

### CODE REVIEW

File: LLPLabelProducer/plugins/LLPGenDecayInfoProducer.cc [hunk 1]
```diff
@@ -149,6 +149,16 @@ LLPGenDecayInfoProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSet
                             {
                                 throw cms::Exception("GenParticle relations not properly setup!");
                             }
+
+                            // Handle taus
+                            if (abs(daughter->pdgId) == 15){
+                                // check for leptonic decays
+                                for(size_t i = 0; i < daughter->numberOfDaughters(); ++i) {
+                                    if (abs(daughter->daughter(i)->pdgId()) == 11 or abs(daughter->daughter(i)->pdgId()) == 13){
+                                        daughter = daughter->daughter(i);
```
Comment (matt-komm) [inline LLPLabelProducer/plugins/LLPGenDecayInfoProducer.cc]: not sure if mu/e are direct daughters of taus or if you have to also account for cases when there is an intermediate W boson stored in the event

File: LLPLabelProducer/plugins/LLPLabelProducer.cc [hunk 2]
```diff
@@ -326,19 +327,25 @@ LLPLabelProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
                             if (absId==11 and pt>electronPtThreshold_) nE+=1;
                             if (absId==5 and pt>bPtThreshold_) nB+=1;
                             if ((absId<5 or absId==21) and pt>quarkPtThreshold_) nQ+=1;
-                                
+                            if ((absId==15) and pt>quarkPtThreshold_) nTau++;
```
Comment (matt-komm) [inline LLPLabelProducer/plugins/LLPLabelProducer.cc]: should be tau threshold

File: LLPLabelProducer/plugins/LLPLabelProducer.cc [hunk 3]
```diff
@@ -326,19 +327,25 @@ LLPLabelProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
...
                         }
+                                
                         label.type = llpdnnx::LLPLabel::Type::isLLP_RAD; //default
-                        
-                        if (nMU==0 and nE==0)
+                        if (nTau>=1)
+                        {
+                            if (nQ+nB==0) label.type = llpdnnx::LLPLabel::Type::isLLP_TAU;
+                            if (nQ+nB==1) label.type = llpdnnx::LLPLabel::Type::isLLP_QTAU;
+                            if (nQ+nB>1) label.type = llpdnnx::LLPLabel::Type::isLLP_QQTAU;
```
Comment (matt-komm) [inline LLPLabelProducer/plugins/LLPLabelProducer.cc]: is this 
(x+y)==z 
or 
x+(y==z)
?

File: LLPLabelProducer/plugins/LLPGenDecayInfoProducer.cc (hunk 1 above)
Comment (daiktas) [reply LLPLabelProducer/plugins/LLPGenDecayInfoProducer.cc]: I think the W* is cleaned in the gen particle collection we're using

File: LLPLabelProducer/plugins/LLPLabelProducer.cc (hunk 2 above)
Comment (daiktas) [reply LLPLabelProducer/plugins/LLPLabelProducer.cc]: ok

File: LLPLabelProducer/plugins/LLPLabelProducer.cc (hunk 3 above)
Comment (daiktas) [reply LLPLabelProducer/plugins/LLPLabelProducer.cc]: ok will make unambiguous

**Pull request merged**